src/Core/FeatureFlag/Layer/DotEnvLayer.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\FeatureFlag\Layer;
+
+use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagSettings;
+
+class DotEnvLayer extends AbstractTypeLayer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getTypeName(): string
+    {
+        return FeatureFlagSettings::TYPE_DOTENV;
+    }
+
+    /**
+     * Retrieve the variable name of this feature flag.
+     */
+    public function getVarName(string $featureFlagName): string
+    {
+        return FeatureFlagSettings::PREFIX . strtoupper($featureFlagName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canBeUsed(string $featureFlagName): bool
+    {
+        return isset($_ENV['SYMFONY_DOTENV_VARS']) && str_contains($_ENV['SYMFONY_DOTENV_VARS'], $this->getVarName($featureFlagName));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEnabled(string $featureFlagName): bool
+    {
+        return isset($_ENV[$this->getVarName($featureFlagName)]) && strtoupper($_ENV[$this->getVarName($featureFlagName)]) === 'TRUE';
```

Comment:
This works if you are only editing via this layer but in case someone edits the .env file manually and uses a different writing like `1` it would be considered as false We'd need a way to check falsy or truthy values a bit more generic/adaptable

Something similar to the env processor from Symfony:
```
env(bool:FOO)
Casts FOO to a bool (true values are 'true', 'on', 'yes' and all numbers except 0 and 0.0; everything else is false):
```

Reply:
I believe you can use the `env` method like this `env('HTTP_METHOD_OVERRIDE')->bool()` I searched into the env processor of Symfony seems like they use `filter_var($env, \FILTER_VALIDATE_BOOLEAN)` So I think you could replace with:

```suggestion
        return isset($_ENV[$this->getVarName($featureFlagName)]) && filter_var($_ENV[$this->getVarName($featureFlagName)], \FILTER_VALIDATE_BOOLEAN);
```

And then you should test multiple values in the unit tests to be sure `1` `on` `true` `TRUE` ...

Reply:
Although I wonder how you can unit test this accurately since it's based on the global var `$_ENV`, I wonder id the env array should be injected? 🤔